cassiopeia/data/CassiopeiaTree.py

```diff
@@ -645,9 +737,34 @@ def set_branch_length(self, parent: str, child: str, length: float):
         if length < 0:
             raise CassiopeiaTreeError("Edge length must be positive.")
 
-        self.__network[parent][child]["length"] = length
+        self.__set_branch_length(parent, child, length)
 
-        for (u, v) in self.depth_first_traverse_edges(source=parent):
+        for u, v in self.depth_first_traverse_edges(source=parent):
+            self.__network.nodes[v]["time"] = (
+                self.__network.nodes[u]["time"] + self.__network[u][v]["length"]
+            )
+
+    def set_branch_lengths(self, branch_length_dict: Dict[Tuple[str, str], float]) -> None:
+        """Sets the length of multiple branches on a tree.
+
+        Adjusts the branch length of specified parent-child relationships.
+        This procedure maintains the consistency with the rest of the times in
+        the tree. Namely, by changing branch lengths here, it will change
+        the times of all the nodes in the tree such that the times are 
+        representative of the new branch lengths.
+
+        Args:
+            branch_dict: A dictionary of edges to updated branch lengths
+        """
+        self.__check_network_initialized()
+
+        for edge, length in branch_length_dict.items():
+            u, v = edge[0], edge[1]
+            if length < 0:
+                raise CassiopeiaTreeError("Edge length must be positive.")
+            self.__network[u][v]["length"] = length
```

Comment:
Why don't we use `set_branch_length` here?

Reply:
`set_branch_length` does a consistency operation, so we'd have to do a consistency operation for each branch. It's more efficient for bulk branch setting I think to do just one consistency operation across the entire tree after all the branches are added. I should be using `__set_branch_length` though you are correct